# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: docshell/base/CanonicalBrowsingContext.cpp
# Commit: ab54092b8afc
# Full Hash: ab54092b8afc8e56dd3f8162344a4ce38ec45999
# Author: Niklas Goegge <ngogge@mozilla.com>
# Date: 2021-04-29 09:26:05
# Regressor Bug: 1708150
# File Overlap Count: 1
# Description:
#   Bug 1708150 - Add user activation flag to reload, goBack and goForward r=ckerschb,Gijs,smaug
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D110245
# ==============================================================================

diff -r a9a52f01de13 -r ab54092b8afc docshell/base/CanonicalBrowsingContext.cpp
--- a/docshell/base/CanonicalBrowsingContext.cpp	Wed Apr 28 11:09:38 2021 +0000
+++ b/docshell/base/CanonicalBrowsingContext.cpp	Wed Apr 28 11:26:49 2021 +0000
@@ -920,7 +920,7 @@
 
 void CanonicalBrowsingContext::HistoryGo(
     int32_t aOffset, uint64_t aHistoryEpoch, bool aRequireUserInteraction,
-    Maybe<ContentParentId> aContentId,
+    bool aUserActivation, Maybe<ContentParentId> aContentId,
     std::function<void(int32_t&&)>&& aResolver) {
   if (aRequireUserInteraction && aOffset != -1 && aOffset != 1) {
     NS_ERROR(
@@ -976,7 +976,8 @@
 
   // GoToIndex checks that index is >= 0 and < length.
   nsTArray<nsSHistory::LoadEntryResult> loadResults;
-  nsresult rv = shistory->GotoIndex(index.value(), loadResults, sameEpoch);
+  nsresult rv = shistory->GotoIndex(index.value(), loadResults, sameEpoch,
+                                    aUserActivation);
   if (NS_FAILED(rv)) {
     MOZ_LOG(gSHLog, LogLevel::Debug,
             ("Dropping HistoryGo - bad index or same epoch (not in same doc)"));
@@ -1110,7 +1111,7 @@
 
 void CanonicalBrowsingContext::GoBack(
     const Optional<int32_t>& aCancelContentJSEpoch,
-    bool aRequireUserInteraction) {
+    bool aRequireUserInteraction, bool aUserActivation) {
   if (IsDiscarded()) {
     return;
   }
@@ -1124,19 +1125,19 @@
     if (aCancelContentJSEpoch.WasPassed()) {
       docShell->SetCancelContentJSEpoch(aCancelContentJSEpoch.Value());
     }
-    docShell->GoBack(aRequireUserInteraction);
+    docShell->GoBack(aRequireUserInteraction, aUserActivation);
   } else if (ContentParent* cp = GetContentParent()) {
     Maybe<int32_t> cancelContentJSEpoch;
     if (aCancelContentJSEpoch.WasPassed()) {
       cancelContentJSEpoch = Some(aCancelContentJSEpoch.Value());
     }
     Unused << cp->SendGoBack(this, cancelContentJSEpoch,
-                             aRequireUserInteraction);
+                             aRequireUserInteraction, aUserActivation);
   }
 }
 void CanonicalBrowsingContext::GoForward(
     const Optional<int32_t>& aCancelContentJSEpoch,
-    bool aRequireUserInteraction) {
+    bool aRequireUserInteraction, bool aUserActivation) {
   if (IsDiscarded()) {
     return;
   }
@@ -1150,18 +1151,19 @@
     if (aCancelContentJSEpoch.WasPassed()) {
       docShell->SetCancelContentJSEpoch(aCancelContentJSEpoch.Value());
     }
-    docShell->GoForward(aRequireUserInteraction);
+    docShell->GoForward(aRequireUserInteraction, aUserActivation);
   } else if (ContentParent* cp = GetContentParent()) {
     Maybe<int32_t> cancelContentJSEpoch;
     if (aCancelContentJSEpoch.WasPassed()) {
       cancelContentJSEpoch.emplace(aCancelContentJSEpoch.Value());
     }
     Unused << cp->SendGoForward(this, cancelContentJSEpoch,
-                                aRequireUserInteraction);
+                                aRequireUserInteraction, aUserActivation);
   }
 }
 void CanonicalBrowsingContext::GoToIndex(
-    int32_t aIndex, const Optional<int32_t>& aCancelContentJSEpoch) {
+    int32_t aIndex, const Optional<int32_t>& aCancelContentJSEpoch,
+    bool aUserActivation) {
   if (IsDiscarded()) {
     return;
   }
@@ -1175,13 +1177,14 @@
     if (aCancelContentJSEpoch.WasPassed()) {
       docShell->SetCancelContentJSEpoch(aCancelContentJSEpoch.Value());
     }
-    docShell->GotoIndex(aIndex);
+    docShell->GotoIndex(aIndex, aUserActivation);
   } else if (ContentParent* cp = GetContentParent()) {
     Maybe<int32_t> cancelContentJSEpoch;
     if (aCancelContentJSEpoch.WasPassed()) {
       cancelContentJSEpoch.emplace(aCancelContentJSEpoch.Value());
     }
-    Unused << cp->SendGoToIndex(this, aIndex, cancelContentJSEpoch);
+    Unused << cp->SendGoToIndex(this, aIndex, cancelContentJSEpoch,
+                                aUserActivation);
   }
 }
 void CanonicalBrowsingContext::Reload(uint32_t aReloadFlags) {